lib/spack/spack/config.py

```diff
@@ -151,18 +155,21 @@ def included_scopes(self) -> List["ConfigScope"]:
             includes = self.get_section("include")
             if includes:
                 include_paths = [included_path(data) for data in includes["include"]]
-                for path in include_paths:
-                    included_scope = include_path_scope(path, self)
-                    if not included_scope:
+                for include in include_paths:
+                    included_scopes = include.scopes(self)
+                    if not included_scopes:
                         continue
```

Comment:
Can't we avoid another level of nesting here by getting all the scopes and iterating on them directly?  Then you don't need the changes to the loop body below.  Something like:

```suggestion
                included_scopes = itertools.chain(include.scopes(self) for include in include_paths)
                for include_scope in included_scopes:
```

And make scopes() return `[]` instead of None when there are no scopes (i.e. make it a `List`, not `Optional[List]` -- `None` isn't actually meaningful vs `[]` is it?)

Reply:
Use of `None` versus an empty list seems to be a matter of reviewer preference IME.

Reply:
Using None here complicates the client code, especially if it is not semantically relevant.

Reply:
Done